### DISCUSSION TITLE
Could macos virtualization be ported to M1 iPad?

### DISCUSSION THREAD

**Discussion by jsonmona** [utmapp/UTM#5454] [General]
As far as I understand, macos virtualization is supported by Virtualization.Framework, not QEMU.

On the other hand, M1 iPad with Trollstore (UTM.HV.ipa), UTM uses QEMU which in turn uses HVF. So it does not support macos virtualization.

Could Virtualization.Framework be "ported" (e.g. bundling the one found on mac) to allow macos virtualization on M1 iPad?

**Comment by osy** [comment]
No it can’t be. I’ve looked into it extensively and the framework uses many macos only frameworks. This includes paravirtualizedgraphics which is required for GPU support on macos. Without it, macos is basically unusable.

**Reply by jsonmona** [reply]
Oh. That's a shame. Thank you for the great software!

**Comment by nfzerox** [comment]
Check out: https://github.com/nfzerox/VirtualMacOniPad
